ttnn/cpp/ttnn/operations/moreh/moreh_linear_backward/device/moreh_linear_backward_device_operation.hpp

```diff
@@ -0,0 +1,110 @@
+// SPDX-FileCopyrightText: © 2024 Tenstorrent Inc.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+#pragma once
+
+#include <variant>
+#include <vector>
+
+#include "ttnn/decorators.hpp"
+#include "ttnn/device_operation.hpp"
+#include "ttnn/operations/core/compute_kernel/compute_kernel_config.hpp"
+#include "ttnn/tensor/tensor.hpp"
+#include "ttnn/tensor/types.hpp"
+
+namespace ttnn::operations::moreh::moreh_linear_backward {
+struct MorehBiasAddBackwardOperation {
+    struct operation_attributes_t {
+        const std::vector<bool> are_required_outputs;
+
+        const MemoryConfig input_grad_mem_config;
+        const MemoryConfig weight_grad_mem_config;
+        const MemoryConfig bias_grad_mem_config;
+        const DeviceComputeKernelConfig compute_kernel_config;
+    };
+
+    struct tensor_args_t {
+        const Tensor& output_grad;
+        const Tensor& input;
+        const Tensor& weight;
+        const std::optional<Tensor>& bias;
+        const std::optional<Tensor>& input_grad;
+        const std::optional<Tensor>& weight_grad;
+        const std::optional<Tensor>& bias_grad;
+    };
```

Comment:
This struct seems to only calculate the bias add backward, yet it receives many unnecessary parameters, which is confusing.

Reply:
I have removed unnecessary params in https://github.com/tenstorrent/tt-metal/pull/12812/commits/8dec1ff696bd9948f9366cbee8ec5733a58f2081